tests/platform_tests/test_auto_negotiation.py

```diff
@@ -186,239 +140,212 @@ def check_ports_up(duthost, dut_ports, expect_speed=None):
     success = fanout.set_speed(fanout_port, None)
     pytest_require(success, 'Failed to advertise all speeds on fanout. Fanout: {}, port: {}'.format(fanout, fanout_port))
 
+    if dut_all_speeds_option == SPEEDS_BY_LITERAL:
+        all_speeds = 'all'
+    else:
+        all_speeds = ','.join(duthost.get_supported_speeds(portname))
+
     duthost.shell('config interface autoneg {} enabled'.format(dut_port))
-    duthost.shell('config interface advertised-speeds {} all'.format(dut_port))
+    duthost.shell('config interface advertised-speeds {} {}'.format(dut_port, all_speeds))
 
     logger.info('Wait until all ports are up')
-    wait_result = wait_until(ALL_PORT_WAIT_TIME, 
-                                PORT_STATUS_CHECK_INTERVAL, 
-                                0, 
-                                check_ports_up, 
-                                duthost, 
-                                [portname])
-    pytest_assert(wait_result, 'Some ports are still down')
+    wait_result = wait_until(
+        ALL_PORT_WAIT_TIME,
+        PORT_STATUS_CHECK_INTERVAL,
+        0,
+        check_ports_up,
+        duthost,
+        [portname])
+    pytest_assert(wait_result, 'The port is still down')
 
     # Make sure ports are negotiated to the highest speed
     logger.info('Checking the actual speed is equal to highest speed')
     int_status = duthost.show_interface(command="status")["ansible_facts"]['int_status']
-    supported_speeds = get_supported_speeds_for_port(duthost, dut_port, fanout, fanout_port)
-    logger.info('DUT port = {}, fanout port = {}, supported speeds = {}, actual speed = {}'.format(
-        dut_port,
-        fanout_port,
-        supported_speeds,
-        int_status[dut_port]['speed']
-    ))
-    highest_speed = supported_speeds[-1]
-    actual_speed = int_status[dut_port]['speed'][:-1] + '000'
+    common_supported_speeds = enum_dut_portname_module_fixture['speeds']
+    highest_speed = max(map(lambda p: int(p), common_supported_speeds))
+    actual_speed = int(int_status[dut_port]['speed'][:-1] + '000')
     pytest_assert(actual_speed == highest_speed, 'Actual speed is not the highest speed')
 
-def test_auto_negotiation_dut_advertises_each_speed(enum_dut_portname_module_fixture):
+
+def is_sfp_speed_supported(duthost, if_name, port_speed):
+    pam4_supporting_sfps = [
+        'QSFP56',
+        'QSFP112',
+        'SFP-DD',
+        'OSFP',
+    ]
+    out = duthost.shell('redis-cli -n 6 HGET "TRANSCEIVER_INFO|{}" "type"'.format(if_name))
+    sfp_type = re.search(r'\w?SFP[\d\w+]{0,3}', out["stdout_lines"][0]).group()
+    if not sfp_type:
+        return True
+    out = duthost.shell('redis-cli -n 4 HGET "PORT|{}" "lanes"'.format(if_name))
+    n_lanes = len(out["stdout_lines"][0].split(','))
+    per_lane_speed = int(port_speed) // n_lanes
+    return per_lane_speed <= 25000 or sfp_type in pam4_supporting_sfps
+
+
+def skip_speed_if_unsupported(duthost, if_name, port_speed):
```

Comment:
do we really need this function? Can we just call pytest_require(is_sfp_speed_supported?)